### ISSUE TITLE
CI clippy config does not check all code

### ISSUE THREAD

**Issue by davepacheco** [oxidecomputer/omicron#1769]
In the neverending saga of trying to get the CI build/check arguments right, today I learned that `cargo clippy` takes the same target arguments as `cargo build`.  And since we're not passing any, we're not checking test code, non-default crates, etc.

CI runs `cargo clippy -- -D warnings -A clippy::style`.  A full run of `cargo clippy --all-targets -- -D warnings -A clippy::style` reveals some issues:

```
error: useless conversion to the same type: `proc_macro2::TokenStream`
   --> nexus/db-macros/src/lookup.rs:900:13
    |
900 | /             quote! {
901 | |                 name = "Organization",
902 | |                 ancestors = ["Silo"],
903 | |                 children = [ "Project" ],
...   |
907 | |             }
908 | |             .into(),
    | |___________________^
    |
    = note: `-D clippy::useless-conversion` implied by `-D warnings`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_conversion
help: consider removing `.into()`
    |
900 ~             quote! {
901 +                 name = "Organization",
902 +                 ancestors = ["Silo"],
903 +                 children = [ "Project" ],
904 +                 lookup_by_name = true,
905 +                 soft_deletes = true,
906 +                 primary_key_columns = [ { column_name = "id", rust_type = Uuid } ]
907 ~             },
    |

error: useless conversion to the same type: `proc_macro2::TokenStream`
   --> nexus/db-macros/src/lookup.rs:914:13
    |
914 | /             quote! {
915 | |                 name = "Project",
916 | |                 ancestors = ["Organization"],
917 | |                 children = [ "Disk", "Instance" ],
...   |
921 | |             }
922 | |             .into(),
    | |___________________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_conversion
help: consider removing `.into()`
    |
914 ~             quote! {
915 +                 name = "Project",
916 +                 ancestors = ["Organization"],
917 +                 children = [ "Disk", "Instance" ],
918 +                 lookup_by_name = true,
919 +                 soft_deletes = true,
920 +                 primary_key_columns = [ { column_name = "id", rust_type = Uuid } ]
921 ~             },
    |

error: useless conversion to the same type: `proc_macro2::TokenStream`
   --> nexus/db-macros/src/lookup.rs:928:13
    |
928 | /             quote! {
929 | |                 name = "SiloUser",
930 | |                 ancestors = [],
931 | |                 children = [],
...   |
935 | |             }
936 | |             .into(),
    | |___________________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_conversion
help: consider removing `.into()`
    |
928 ~             quote! {
929 +                 name = "SiloUser",
930 +                 ancestors = [],
931 +                 children = [],
932 +                 lookup_by_name = false,
933 +                 soft_deletes = true,
934 +                 primary_key_columns = [ { column_name = "id", rust_type = Uuid } ]
935 ~             },
    |

error: useless conversion to the same type: `proc_macro2::TokenStream`
   --> nexus/db-macros/src/lookup.rs:942:13
    |
942 | /             quote! {
943 | |                 name = "UpdateAvailableArtifact",
944 | |                 ancestors = [],
945 | |                 children = [],
...   |
953 | |             }
954 | |             .into(),
    | |___________________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_conversion
help: consider removing `.into()`
    |
942 ~             quote! {
943 +                 name = "UpdateAvailableArtifact",
944 +                 ancestors = [],
945 +                 children = [],
946 +                 lookup_by_name = false,
947 +                 soft_deletes = false,
948 +                 primary_key_columns = [
949 +                     { column_name = "name", rust_type = String },
950 +                     { column_name = "version", rust_type = i64 },
951 +                     { column_name = "kind", rust_type = UpdateArtifactKind }
952 +                 ]
953 ~             },
    |

error: useless conversion to the same type: `proc_macro2::TokenStream`
   --> nexus/db-macros/src/lib.rs:437:13
    |
437 | /             quote! {
438 | |                 #[derive(Resource)]
439 | |                 struct MyTarget {
440 | |                     identity: MyTargetIdentity,
...   |
444 | |             }
445 | |             .into(),
    | |___________________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_conversion
help: consider removing `.into()`
    |
437 ~             quote! {
438 +                 #[derive(Resource)]
439 +                 struct MyTarget {
440 +                     identity: MyTargetIdentity,
441 +                     name: String,
442 +                     is_cool: bool,
443 +                 }
444 ~             },
    |

error: useless conversion to the same type: `proc_macro2::TokenStream`
   --> nexus/db-macros/src/lib.rs:459:13
    |
459 | /             quote! {
460 | |                 #[derive(Resource)]
461 | |                 #[diesel(table_name)]
462 | |                 struct MyTarget {
...   |
467 | |             }
468 | |             .into(),
    | |___________________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_conversion
help: consider removing `.into()`
    |
459 ~             quote! {
460 +                 #[derive(Resource)]
461 +                 #[diesel(table_name)]
462 +                 struct MyTarget {
463 +                     identity: MyTargetIdentity,
464 +                     name: String,
465 +                     is_cool: bool,
466 +                 }
467 ~             },
    |

error: useless conversion to the same type: `proc_macro2::TokenStream`
   --> nexus/db-macros/src/lib.rs:482:13
    |
482 | /             quote! {
483 | |                 #[derive(Resource)]
484 | |                 #[diesel(table_name = foo)]
485 | |                 enum MyTarget {
...   |
489 | |             }
490 | |             .into(),
    | |___________________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_conversion
help: consider removing `.into()`
    |
482 ~             quote! {
483 +                 #[derive(Resource)]
484 +                 #[diesel(table_name = foo)]
485 +                 enum MyTarget {
486 +                     Foo,
487 +                     Bar,
488 +                 }
489 ~             },
    |

error: useless conversion to the same type: `proc_macro2::TokenStream`
   --> nexus/db-macros/src/lib.rs:503:13
    |
503 | /             quote! {
504 | |                 #[derive(Resource)]
505 | |                 #[diesel(table_name = my_target)]
506 | |                 struct MyTarget {
...   |
510 | |             }
511 | |             .into(),
    | |___________________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_conversion
help: consider removing `.into()`
    |
503 ~             quote! {
504 +                 #[derive(Resource)]
505 +                 #[diesel(table_name = my_target)]
506 +                 struct MyTarget {
507 +                     name: String,
508 +                     is_cool: bool,
509 +                 }
510 ~             },
    |

error: useless conversion to the same type: `proc_macro2::TokenStream`
   --> nexus/db-macros/src/lib.rs:526:13
    |
526 | /             quote! {
527 | |                 #[derive(Resource)]
528 | |                 #[diesel(table_name = my_target)]
529 | |                 struct MyTarget {
...   |
534 | |             }
535 | |             .into(),
    | |___________________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_conversion
help: consider removing `.into()`
    |
526 ~             quote! {
527 +                 #[derive(Resource)]
528 +                 #[diesel(table_name = my_target)]
529 +                 struct MyTarget {
530 +                     identity: MyTargetIdentity,
531 +                     name: String,
532 +                     is_cool: bool,
533 +                 }
534 ~             },
    |

error: could not compile `db-macros` due to 9 previous errors
warning: build failed, waiting for other jobs to finish...
error: useless use of `vec!`
   --> oximeter/oximeter/src/histogram.rs:657:39
    |
657 |         let mut hist = Histogram::new(&vec![0, 10, 20]).unwrap();
    |                                       ^^^^^^^^^^^^^^^^ help: you can use a slice directly: `&[0, 10, 20]`
    |
    = note: `-D clippy::useless-vec` implied by `-D warnings`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_vec

error: useless use of `vec!`
   --> oximeter/oximeter/src/histogram.rs:785:28
    |
785 |             Histogram::new(&vec![0, -10, 1]).is_err(),
    |                            ^^^^^^^^^^^^^^^^ help: you can use a slice directly: `&[0, -10, 1]`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_vec

error: useless use of `vec!`
   --> oximeter/oximeter/src/histogram.rs:797:39
    |
797 |         let mut hist = Histogram::new(&vec![0.0, 1.0]).unwrap();
    |                                       ^^^^^^^^^^^^^^^ help: you can use a slice directly: `&[0.0, 1.0]`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_vec

error: using `clone` on type `types::Cumulative<i64>` which implements the `Copy` trait
   --> oximeter/oximeter/src/types.rs:717:44
    |
717 |         let measurement = Measurement::new(datum.clone());
    |                                            ^^^^^^^^^^^^^ help: try removing the `clone` call: `datum`
    |
    = note: `-D clippy::clone-on-copy` implied by `-D warnings`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#clone_on_copy

error: could not compile `oximeter` due to 4 previous errors
```

**Comment by ryaeng** [comment]
I've been working through a lot of these errors. Came across this one and was wondering if anyone could recommend a solution.

error: very complex type used. Consider factoring parts into `type` definitions
   --> common/src/api/external/http_pagination.rs:627:10
    |
627 |       ) -> (
    |  __________^
628 | |         PaginationParams<S, PageSelector<S, S::MarkerValue>>,
629 | |         PaginationParams<S, PageSelector<S, S::MarkerValue>>,
630 | |     )
    | |_____^
    |
    = note: `-D clippy::type-complexity` implied by `-D warnings`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#type_complexity

**Comment by ryaeng** [comment]
Looks like mocking an instance for the sled-agent has an argument list that's too long. Any suggestions on cleaning up the inputs?

`    pub Instance {
        pub fn new(
            log: Logger,
            id: Uuid,
            vnic_allocator: VnicAllocator,
            underlay_addr: Ipv6Addr,
            port_allocator: OptePortAllocator,
            initial: InstanceHardware,
            nexus_client: Arc<NexusClient>,`

**Comment by davepacheco** [comment]
The line numbers here don't appear to match "main".  Are you talking about this:
https://github.com/oxidecomputer/omicron/blob/4886bf8a153f2b65b24ad7f8b87233368e5cbbbb/sled-agent/src/instance.rs#L419-L424

That differs from what's in your gist, though.  This function currently has only six args, which is under Clippy's threshold.  It's not the same code as your error because it's called `lazy_nexus_client` here.

**Comment by ryaeng** [comment]
@davepacheco My fork was out of sync. Fixed it. No longer a problem.

Submitted the PR.

**Comment by davepacheco** [comment]
Fixed by #2183
